streamer/pulp/streamer/server.py

```diff
@@ -40,92 +41,30 @@
-        """
-        Forward a subset of the HTTP headers received from the upstream server as
-        well as set the cache-control header to the value specified by the streamer
-        configuration file. This header is used by clients to determine how to
-        cache the response.
-
-        :param report: The download report for this request.
-        :type  report: nectar.report.DownloadReport
-        """
-        for header_key, header_value in report.headers.items():
-            if header_key.lower() not in HOP_BY_HOP_HEADERS:
-                self.request.setHeader(header_key, header_value)
-
-        max_age = {'max_age': self.streamer_config.get('streamer', 'cache_timeout')}
-        cache_header = 'public, s-maxage=%(max_age)s, max-age=%(max_age)s' % max_age
-        self.request.setHeader('Cache-Control', cache_header)
+class DownloadListener(AggregatingEventListener):
+    """
+    Nectar download listener.
+    """
 
     def download_failed(self, report):
         """
-        Perform cleanup on failed downloads. Specifically, Nectar does not download
-        any content if the server returns a non-200 code. This is problematic because
-        it will return headers to the client which include a Content-Length that is
-        likely not 0, since most servers provide pages detailing the problem that
-        occurred.
-
-        :param report: The download report for this request.
-        :type  report: nectar.report.DownloadReport
-        """
-        # Currently Nectar returns headers with a content-length even
-        # when it doesn't download anything.
-        self.request.setHeader('Content-Length', '0')
-        if 'response_code' in report.error_report:
-            self.request.setResponseCode(report.error_report['response_code'])
-        else:
-            # Nectar doesn't give us a good way to know exactly went wrong; return
-            # a generic HTTP 503 and hope Nectar logged enough to be useful
-            self.request.setResponseCode(SERVICE_UNAVAILABLE)
-
-    def download_succeeded(self, report):
-        """
-        If the download was successful, add a deferred download entry.
+        Log download failures.
 
-        :param report: The download report for this request.
+        :param report: The download report.
         :type  report: nectar.report.DownloadReport
         """
-        if not self.pulp_request:
-            try:
-                download = model.DeferredDownload(
-                    unit_id=self.catalog_entry.unit_id,
-                    unit_type_id=self.catalog_entry.unit_type_id
-                )
-                download.save()
-            except NotUniqueError:
-                # There's already an entry for this unit.
-                pass
+        super(DownloadListener, self).download_failed(report)
+        code = report.error_report.get('response_code', NOT_FOUND)
+        logger.info(_('Download failed [{code}]: {url}').format(code=code, url=report.url))
```

Comment:
what if we move this to debug level. I am afraid it would pollute info level with a lot of messages.